text/XXXX-storage_command.md

```diff
@@ -0,0 +1,178 @@
+  qri purge [REMOTE] DATASET [flags]
+
+Examples:
+  # drop
+
+Flags:
+  -a, --all               purge all versions, overrides revisions
+  -r, --revisions         purge a number of versions from the latest commit
+  -t, --tail              purge stored versions starting from the earliest
+
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The purpose of stats are to answer a few primary questions. All of them deal with a _set of blocks_. 
+
+
+The foundation for all of this work should be a `dag.Info` & `dag.Manifest`. Info provides the block size information, and manifests provide the . `dag.Info` currently looks like this:
+
+```go
+// Info is os.FileInfo for dags: a struct that describes important
+// details about a graph. Info builds on a manifest.
+//
+// when being sent over the network, the contents of Info should be considered gossip,
+// as Info's are *not* deterministic. This has important implications
+// Info should contain application-specific info about a datset
+type Info struct {
+	// Info is built upon a manifest
+	Manifest *Manifest      `json:"manifest"`
+	Labels   map[string]int `json:"labels,omitempty"` // sections are lists of logical sub-DAGs by positions in the nodes list
+	Sizes    []uint64       `json:"sizes,omitempty"`  // sizes of nodes in bytes
+}
+```
+
+Here's the current Manifest data structure:
+```go
+type Manifest struct {
+  Links [][2]int `json:"links"` // links between nodes
+  Nodes []string `json:"nodes"` // list if CIDS contained in the DAG
+}
+```
+
+Manifests assume they describe one graph with a single root. To represent multiple graphs in the same manifest, we'll need to make one change:
+
+```go
+type Manifest struct {
+  Graphs [][][2]int `json:"links"` // links between nodes
+  Nodes []string `json:"nodes"` // list if CIDS contained in the DAG
+}
+```
+
+Calculating storage states
+1. Load logbook info for a dataset
+2. Build a `dag.Manifest` of all stored roots, keeping a slice of block sizes while you're at it
+3. Turn that into a `dag.Info`
+4. Call methods on `dag.Info` that produce storage stats
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+<!-- - Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not choosing them?
+- What is the impact of not doing this? -->
+
+# Prior art
+[prior-art]: #prior-art
+
+
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+## Knowing which datasets depend on a block
+One of the problems this RFC doesn't tackle properly is _knowing_ if removing a block will free up storage space, because we aren't leveraging pin reference counting. In an ideal world we'd have a low-level command like this:
+
+```
+qri storage deps: for a given block, show which dataset versions depend on that that block
```

Comment:
Do we also have, or at least want to have, de-duplication across multiple datasets based on these blocks. In that case reference counting is a must (and opens us up to a whole slew of weird race conditions with peers unless we guard with some sort of replication strategy?)

Reply:
b/c of ipfs we have natural dedupe on identical blocks added to the ipfs store from any datasets added. So v good point about clean up, need to ensure that we aren't depending on these blocks in other contexts before removing, although this feels like a problem ipfs would have had to tackle when 'unpinning' already.

Reply:
truth. the pin API maintains reference counts for each block under the hood. That's why the command to collect & delete 0 reference-count blocks is called `ipfs repo gc` 😄